server/server.js

```diff
@@ -19,41 +23,45 @@ const distPath = path.resolve(__dirname, '..', 'dist');
 const store = makeStore();
 const RoutingContext = createFactory(RoutingContextComponent);
 const Provider = createFactory(ProviderComponent);
+const verify = (token, callback) => {
+  return jwtoken.verify(token, 'secret', {}, callback);
+};
 
 const MongoDB = mongoose.connect(process.env.MONGOLAB_URI || config.developmentDB).connection;
 MongoDB.on('error', (err) => {
   console.log('Failed to connect go Mongo DB');
   console.log(err.message);
 });
-
 MongoDB.once('open', () => {
   console.log('Connected to Mongo DB');
 });
 
-console.log('Server running...');
-
 app.use(express.static(distPath));
+app.use(cookieParser());
 app.use((req, res) => {
   const location = createLocation(req.url);
+  const posToken = req.cookies._posToken;
 
   match({ routes, location }, (err, redirectLocation, renderProps) => {
-    if (err) {
-      console.error(err);
-      return res.status(500).end('Internal server error');
-    }
-
-    if (!renderProps) return res.status(404).end('Not found');
-
-    const InitialComponent = Provider(
-      { store },
-      RoutingContext(renderProps)
-    );
-
-    const initialState = store.getState();
-
-    const componentHTML = renderToString(InitialComponent);
-
-    const HTML = `
+    verify(posToken, (err, token) => {
+      if (err) {
+        console.error(err);
+        return res.redirect(302, '/');
```

Comment:
http status wise, how do you think we should handle it? Using a redirect when it's a 'user not authorized' sounds iffy.